### PR TITLE
kms - support for R formulas for keras_sequential_model()

### PR DESCRIPTION
**Pull request by rdrr1990**
This PR adds `kms()`, which builds and fits a `keras_sequential_model` (or, optionally, allows the user to pass a compiled model) based on an R `formula`. `kms` accepts a number of parameters (like `loss`,`optimizer`, `seed`, and `Nepochs`) and splits the data into test and training sparse matrices (from library `Matrix`). `kms` returns a single object of class `kms_fit` with predictions, a confusion matrix (lightly formatted by the only other new function, `confusion`), as well as details about the function call.  

`kms`, by default, builds a simple dense neural net (with dropout after each layer). `kms` allows users to input a list that controls the number of layers and the respective dropout rates. For more complex models like `lstm`, it is recommended that users pass in a compiled model. 

The vignette shows basics and an additional example shows kms' utility on raw data [here](http://bit.ly/2CcirgY). 

Please let me know if you have any questions or concerns. `keras` is a wonderful addition to R and hopefully `kms` can help users quickly navigate data cleaning and focus on model building and selection. Happy of course to make edits!

### CODE REVIEW

**Comment by jjallaire** [comment]
Thanks for this, very nice!

This represents one of many possible higher level interfaces to Keras from R, and I'd expect (and hope!) to see others emerge over time. Since we want to encourage a diversity of these, I don't want to encode one particular interface into the keras package right now (obviously if one got huge momentum and was clearly the consensus best I'd consider incorporating it).

That said, if you publish this as a standalone package on CRAN I would certainly link to it from the Keras for R website and also invite you to submit a guest post about it on the TensorFlow for R blog: tensorflow.rstudio.com/blog

**Comment by rdrr1990** [comment]
Sounds good. I had wondered if a standalone package might be better. Think I'll call it kerasFormula so it's easy to find on CRAN. I'll be in touch once it's up.

**Comment by jjallaire** [comment]
Sounds good, thanks!

**Comment by jjallaire** [comment]
I might suggest `kerasformula` as the package name as I think using one contiguous word is more consistent with the snake case used in the package. I also think that package names which start with a lowercase letter then include an uppercase one look odd (I don't feel this way about variable names in camel case code, just top level objects like modules/packages/classes).

For reference, here are all of the "shiny" derivative package names on CRAN:

```
shiny.semantic
shinyAce
shinyaframe
shinybootstrap2
shinyBS
shinycssloaders
shinydashboard
shinyDND
shinyFeedback
shinyFiles
shinyHeatmaply
ShinyImage
ShinyItemAnalysis
shinyjqui
shinyjs
shinyKGode
shinyLP
shinymaterial
shinyRGL
shinyShortcut
shinystan
ShinyTester
shinythemes
shinyTime
shinytoastr
shinyTree
shinyWidgets
```

I personally think the ones without the uppercase look better, but obviously all approaches work in their own way and can co-exist. Just my 2 cents...

**Comment by rdrr1990** [comment]
works for me (didn't have any strong feelings on that one), thx!

**Comment by rdrr1990** [comment]
Here is a link to the `kerasformula` repo: https://github.com/rdrr1990/kerasformula

**Comment by rdrr1990** [comment]
After a bit of holiday delay, `kerasformula` is heading to `CRAN`. I will send a demo your way soon.

**Comment by jjallaire** [comment]
That's great news!

**Comment by rdrr1990** [comment]
How would something like this be for the tensoflow for R/RStudio site?

https://github.com/rdrr1990/kerasformula/blob/master/mohanty_kerasformula.md

**Comment by jjallaire** [comment]
Hi @rdrr1990, that looks like a great start! We'll publish a version of this on the TensorFlow for R blog.

@dfalbel Has helped with writing a number of posts on the blog (and is also responsible for nearly all of the Keras examples on our site) and has agreed to help with editing / refining your article. I suggest that we start a repo where all of us have commit access and we can iterate on it there.

**Comment by rdrr1990** [comment]
Sounds great, I'll set that up
